console/src/components/Services/Events/Common/Components/EventsTable.tsx

```diff
@@ -86,16 +89,22 @@ const EventsTable: React.FC<Props> = props => {
 
   const changePage = (page: number) => {
     if (filterState.offset !== page * filterState.limit) {
+      const offset = page * pgSize;
+      setCurrentPage(page);
       runQuery({
-        offset: page * filterState.limit,
+        offset,
+        limit: offset + pgSize,
```

Comment:
Note to self to change this to use the `offset` alone and keep the `limit` as a constant. the slicing can be done via, `offset` and `offset + pagesize`